### PR TITLE
service/block: Remove block fetching capability from `BlockService`

### LINKED ISSUE #251: service/block, service/header: Move `ExtendedHeader` creation from `BlockService` to `HeaderService`
Ref #244 
Related to #214 

TODO: 
- [ ] refactor `BlockService` so that it no longer subscribes to new block events from core
- [ ] refactor `HeaderService` so that it is able to subscribe to new block events from core after it is finished syncing headers (#327)

### PR DESCRIPTION
**Pull request by renaynay**
This PR is the first step necessary to resolve #251 (it resolves the first TODO).

### CODE REVIEW

**Comment by Wondertan** [comment]
Btw, there should be a changelog entry for this one.

File: node/services/service.go [hunk 1]
```diff
@@ -78,11 +77,9 @@ func HeaderStore(ds datastore.Batching) (header.Store, error) {
 // BlockService constructs new block.Service.
 func BlockService(
 	lc fx.Lifecycle,
-	fetcher *core.BlockFetcher,
 	store ipld.DAGService,
-	broadcaster header.Broadcaster,
 ) *block.Service {
-	service := block.NewBlockService(fetcher, store, broadcaster)
+	service := block.NewBlockService(store)
```
Comment (adlerjohn) [inline node/services/service.go:82]: You're removing the fetcher and broadcaster here, but aren't they needed for fetching and broadcasting blocks, not headers?

File: utils/erasure.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+)
+
+// ExtendBlock erasure codes the given raw block's data and returns the
+// erasure coded block data upon success.
+func ExtendBlock(block *core.Block, dag format.DAGService) (*rsmt2d.ExtendedDataSquare, error) {
+	namespacedShares, _ := block.Data.ComputeShares()
+	shares := namespacedShares.RawShares()
+
+	// create nmt adder wrapping batch adder
+	batchAdder := ipld.NewNmtNodeAdder(context.Background(), format.NewBatch(context.Background(), dag))
```
Comment (adlerjohn) [inline utils/erasure.go:24]: Why was this changed from `ce.shareStore` to `dag format.DAGService`?

File: utils/erasure.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+	// compute extended square
+	return rsmt2d.ComputeExtendedDataSquare(shares, rsmt2d.NewRSGF8Codec(), tree.Constructor)
+}
+
+// squareSize64 computes the square size as uint64 from
+// the given length of shares.
+func squareSize64(length int) uint64 {
+	return uint64(math.Sqrt(float64(length)))
+}
```
Comment (adlerjohn) [inline utils/erasure.go:38]: Is there a reason the TODO to un-dup this was removed?

**Comment by Wondertan** [comment]
@adlerjohn, note that this PR is a part of a bigger goal and changes here might not make sense in isolation

**Comment by adlerjohn** [comment]
@Wondertan I gathered that much 😂. So long as the noted peculiarities are expected I have no issue with them temporarily existing.

File: node/services/service.go (hunk 1 above)
Comment (renaynay) [reply node/services/service.go:82]: No, block reconstruction will be done by fetching shares via `ShareService` instead and reconstructed / stored within `BlockService`.

File: utils/erasure.go (hunk 3 above)
Comment (renaynay) [reply utils/erasure.go:38]: I've de-duplicated it as much as possible (besides the IPLD package to avoid circular dependency) which will eventually be separated out into its own repo later on.

**Review by adlerjohn — APPROVED** [review]

File: utils/erasure.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
+package utils
```
Comment (Wondertan) [inline utils/erasure.go:1]: Why utils? 

I like the approach of making things reusable but in a modular domain-specific way, not a monolithic way. And we are already going with the former.

The latter:
services/
    header_service.go
    share_service.go

The former: 
header/
    service.go
share/
    service.go

Same with utils. Having one big folder for all the utils is monolithic and prone to dependency cycles, as it depends on our data structures.

If this makes sense to you, pls move you the helper func inside a respective module/domain package.

**Review by adlerjohn — COMMENTED** [review]
conflicts

**Comment by renaynay** [comment]
@adlerjohn closing this PR in favour of #317
